ambari-server/src/main/java/org/apache/ambari/server/events/MessageEmitter.java

```diff
@@ -269,9 +193,135 @@ protected void emitExecutionCommandToHost(ExecutionCommandEvent event) throws Ho
   @Subscribe
   public void onHostRegister(HostRegisteredEvent hostRegisteredEvent) {
     Long hostId = hostRegisteredEvent.getHostId();
-    if (!messagesToEmit.containsKey(hostId)) {
-      messagesToEmit.put(hostId, new LinkedBlockingQueue<>());
-      monitors.put(hostId, monitorsExecutor.submit(new MessagesToEmitMonitor(hostId)));
+    messagesToEmit.computeIfAbsent(hostId, id -> new LinkedBlockingQueue<>());
+  }
+
+  private class MessagesToEmitMonitor implements Runnable {
+
+    private AtomicBoolean anyActionPerformed = new AtomicBoolean(false);
+
+    @Override
+    public void run() {
+      while (true) {
+        anyActionPerformed.set(false);
```

Comment:
Seems action is performed here vs not.

Reply:
Rechecked, this is actually correct behavior.
We sleep if we queues were all empty to wait a bit and free some cycles as it is unlikely that something will appear in <1ms (task scheduling is very rare operation comparing to cluster lifetime).